qubes-rpc/qvm-copy-to-vm.kde

```diff
@@ -19,25 +19,33 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #
 #
+if type kdialog 2> /dev/null; then
+    kdlg=1
+    else kdlg=0
+fi
 
-VM=$(kdialog -inputbox "Enter the VM name to send files to:")
-if [ X$VM = X ] ; then exit 0 ; fi
+if [ $kdlg -eq 1 ] ; then
+    VM=$(kdialog -inputbox "Enter the VM name to send files to:")
+    if [ X$VM = X ] ; then exit 0 ; fi
 
-SIZE=$(du --apparent-size -c -- "$@" 2> /dev/null | tail -1 | cut -f 1)
-REF=$(kdialog --progressbar "Copy progress")
-qdbus $REF org.freedesktop.DBus.Properties.Set "" maximum $SIZE
+    SIZE=$(du --apparent-size -c -- "$@" 2> /dev/null | tail -1 | cut -f 1)
+    REF=$(kdialog --progressbar "Copy progress")
+    qdbus $REF org.freedesktop.DBus.Properties.Set "" maximum $SIZE
 
-export PROGRESS_TYPE=gui
+    export PROGRESS_TYPE=gui
 
-/usr/lib/qubes/qrexec-client-vm $VM qubes.Filecopy \
-	/usr/lib/qubes/qfile-agent "$@" |
-(while read sentsize ; do
-	CURRSIZE=$(($sentsize/1024))
-	qdbus $REF  org.freedesktop.DBus.Properties.Set "" value $CURRSIZE
-done)
+    /usr/lib/qubes/qrexec-client-vm $VM qubes.Filecopy \
+        /usr/lib/qubes/qfile-agent "$@" |
+    (while read sentsize ; do
+        CURRSIZE=$(($sentsize/1024))
+        qdbus $REF  org.freedesktop.DBus.Properties.Set "" value $CURRSIZE
+    done)
 
-qdbus $REF close
-# we do not want a dozen error messages, do we
-# if ! [ "x"$agentstatus = xDONE ] ; then
-# 	kdialog --sorry 'Abnormal file copy termination; see /var/log/qubes/qrexec.xid.log in dom0 for more details'
-# fi
+    qdbus $REF close
+    # we do not want a dozen error messages, do we
+    # if ! [ "x"$agentstatus = xDONE ] ; then
+    # 	kdialog --sorry 'Abnormal file copy termination; see /var/log/qubes/qrexec.xid.log in dom0 for more details'
+    # fi
+else
+    /usr/lib/qubes/qvm-copy-to-vm.gnome "$@" &
```

Comment:
Even if exit code is not useful (not really sure about that - but nautilus/dolphin _may_ use it to report possible failure), running it in background may result in unexpected results.